File: testdata/swift/expected.swift
import Foundation

/// Documentation comment for the class
/// This is a multi-line documentation comment
class SwiftExample {
    var name: String = "example"
    
    /* Multi-line comment
       spanning multiple lines
       with more details */
    private var count: Int = 0
    
    /**
     * JavaDoc style documentation comment
     * @param name The name parameter
     * @returns Void
     */
    func setName(_ name: String) {
        self.name = name
    }
    
    // TODO: Implement this method
    func increment() {
        /* Block comment */ count += 1
    }
    
    /// Single line doc comment
    func getValue() -> Int {
        return count
    }
} 

File: testdata/cpp/expected.cpp
#include <iostream>
#include <vector>
#include <memory>

// TODO: Implement better error handling
// FIXME: Memory leak in constructor
// NOTE: This class is thread-safe

/*
   This is a multi-line comment
   that should be preserved
*/

/**
 * Documentation comment for the class
 * @param value Initial value
 */
class Example {
private:
    int value;

public:
    Example(int val) : value(val) {
        std::cout << "Creating Example with value: " << val << std::endl;
    }

    int getValue() const {
        return value;
    }

    void setValue(int newVal) {
        value = newVal;
    }

    /* Block comment in method */
    void processData() {
        std::vector<int> data = {1, 2, 3, 4, 5};

        for (const auto& item : data) {
            std::cout << item << " ";
        }
        std::cout << std::endl;
    }
};

int main() {
    Example ex(42);

    ex.processData();

    int val = ex.getValue();
    std::cout << "Value: " << val << std::endl;

    return 0;
}